AirCasting/SessionViews/SessionCartViewModel/MeasurementsViewModel.swift

```diff
@@ -0,0 +1,67 @@
+// Created by Lunar on 11/09/2021.
+//
+
+import Foundation
+import CoreLocation
+import CoreData
+
+protocol SyncMeasurements {
+    func syncMeasurements(for session: SessionEntity) throws
+}
+
+class CardMeasurementsViewModel: ObservableObject, SyncMeasurements {
+    
+    @Published var showLoadingIndicator = true
+    private let sessionDownloader: MeasurementsDownloadable
+    private var measurementStreamStorage: MeasurementStreamStorage?
+    private var task: Cancellable?
+    
+    init(measurementStreamStorage: MeasurementStreamStorage?, sessionDownloader: MeasurementsDownloadable) {
+        self.sessionDownloader = sessionDownloader
+        self.measurementStreamStorage = measurementStreamStorage
+    }
+    
+    func syncMeasurements(for session: SessionEntity) {
+        guard let measurementStreamStorage = measurementStreamStorage else { return }
+        showLoadingIndicator = true
+        
+        task = sessionDownloader.downloadSessionWithMeasurement(uuid: session.uuid) { [weak self] result in
```

Comment:
thought: should we cancel download if user collapses the cell while download in progress?

Reply:
Dunno, I'll ask

Reply:
Nope, they don't do that so I gues we won't be doing that either.